src/thread_task.rs

```diff
@@ -840,20 +910,18 @@ pub unsafe fn rav1d_worker_task(c: &Rav1dContext, task_thread: Arc<Rav1dTaskCont
 
                             // if not bottom sbrow of tile, this task will be re-added
                             // after it's finished
-                            if check_tile(t_idx, &f, &fc.task_thread, (c.fc.len() > 1) as c_int)
-                                == 0
-                            {
+                            if check_tile(&f, &fc.task_thread, &t, (c.fc.len() > 1) as c_int) == 0 {
                                 break 'found (fc, t_idx, prev_t);
                             }
                         } else if t.recon_progress != 0 {
                             let f = fc.data.try_read().unwrap();
                             let p = t.type_0 == TaskType::EntropyProgress;
                             let error = fc.task_thread.error.load(Ordering::SeqCst);
-                            if !(fc.task_thread.done[p as usize].load(Ordering::SeqCst) == 0
-                                || error != 0)
-                            {
-                                unreachable!();
+                            let done = fc.task_thread.done[p as usize].load(Ordering::SeqCst);
+                            if !(done == 0 || error != 0) {
+                                dbg!(done, error);
                             }
+                            assert!(done == 0 || error != 0);
```

Comment:
Did you mean to leave in this `dbg!` print? If so it'd probably be better to print out the values in the `assert!` rather than a separate `dbg!`.